simexpal/base.py

```diff
@@ -247,12 +262,27 @@ def revisions_for_experiment(exp_info):
 							else:
 								reps = range(0, 1)
 							for rep in reps:
-								yield (Experiment(self, exp_info, revision, variation), instance, rep)
-
-		key = lambda t: (t[0].name, t[0].revision.name if t[0].revision is not None else '_none',
-						 [sub_var.name for sub_var in t[0].variation], t[1].shortname, t[2])
-		for experiment, instance, rep in self._expand_matrix(extract, key=key):
-			yield Run(self, experiment, instance, rep)
+								if (exp_info, revision, variation) in self._experiments:
+									yield (self._experiments[(exp_info, revision, variation)], instance, rep)
+								else:
+									self._experiments[(exp_info, revision, variation)] = Experiment(self, exp_info, revision, variation)
+									yield (self._experiments[(exp_info, revision, variation)], instance, rep)
+			self._experiments_discovered = True
+
+		if self._runs_discovered:
+			for experiment in self._runs:
+				for instance, rep in self._runs[experiment]:
+					yield Run(self, experiment, instance, rep)
```

Comment:
Make `_runs` a single-level dict.

Reply:
No, I mean a `dict()` that just has `Run` objects as values (and tuples of Experiment/Instance/Repetition as keys, I guess).

Reply:
done